examples/ucp_client_server.c

```diff
@@ -59,6 +67,16 @@ typedef enum {
 } send_recv_type_t;
 
 
+typedef struct data_meta {
+    int              is_server;
+    send_recv_type_t send_recv_type;
+    ucp_datatype_t   data_type;
+    size_t           buffer_size;
```

Comment:
Here is not clear. IMO, contig is similar as single iov. So use buffer_size to indicate buffer size for contig, but iov number for IOV is not good even if the value has the same type. Please define more clear, for example:

```int                     is_server;
send_recv_type_t send_recv_type;
ucp_datatype_t    data_type;
void                     *buffer;
size_t                   contig_buffer_size;
size_t                   iov_num;
size_t                   *iov_sizes;
```
You can use union to save memory, but don't use the same variable to indicate different concepts.